core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -925,6 +925,25 @@ oppia.factory('explorationStatesService', [
       var conversionFunction = BACKEND_CONVERSIONS[backendName];
       return conversionFunction(frontendValue);
     };
+    
+    var finallyDelete = function(deleteStateName){
+      _states.deleteState(deleteStateName);
+
+      changeListService.deleteState(deleteStateName);
+
+      if (editorContextService.getActiveStateName() === deleteStateName) {
+        editorContextService.setActiveStateName(
+              explorationInitStateNameService.savedMemento);
+      }
+
+      $location.path('/gui/' + editorContextService.getActiveStateName());
+      $rootScope.$broadcast('refreshGraph');
+          // This ensures that if the deletion changes rules in the current
```

Comment:
@himanshu-dixit functionality for moving questions is not included in this PR. It will need a separate PR of its own. But now it can be done as Edit mode is created in this PR.